beacon-chain/sync/rpc_beacon_blocks_by_range_test.go

```diff
@@ -336,3 +337,117 @@ func TestRPCBeaconBlocksByRange_RPCHandlerRateLimitOverflow(t *testing.T) {
 		assert.Equal(t, expectedCapacity, remainingCapacity, "Unexpected rate limiting capacity")
 	})
 }
+
+func TestValidateRangeRequests(t *testing.T) {
+	slotsSinceGenesis := 1000
+	r := &Service{chain: &chainMock.ChainService{
+		Genesis: time.Now().Add(time.Second * time.Duration(-slotsSinceGenesis*int(params.BeaconConfig().SecondsPerSlot))),
+	}}
+
+	tests := []struct {
+		name          string
+		req           *pb.BeaconBlocksByRangeRequest
+		expectedError string
+		errorToLog    string
+	}{
+		{
+			name: "Zero Count",
+			req: &pb.BeaconBlocksByRangeRequest{
+				Count: 0,
+				Step:  1,
+			},
+			expectedError: reqError,
+			errorToLog:    "validation did not fail with bad count",
+		},
+		{
+			name: "Over limit Count",
+			req: &pb.BeaconBlocksByRangeRequest{
+				Count: params.BeaconNetworkConfig().MaxRequestBlocks + 100,
+				Step:  1,
+			},
+			expectedError: reqError,
+			errorToLog:    "validation did not fail with bad count",
+		},
+		{
+			name: "Correct Count",
+			req: &pb.BeaconBlocksByRangeRequest{
+				Count: params.BeaconNetworkConfig().MaxRequestBlocks - 100,
+				Step:  1,
+			},
+			errorToLog: "validation failed with correct count",
+		},
+		{
+			name: "Zero Step",
+			req: &pb.BeaconBlocksByRangeRequest{
+				Step:  0,
+				Count: 1,
+			},
+			expectedError: reqError,
+			errorToLog:    "validation did not fail with bad step",
+		},
+		{
+			name: "Over limit Step",
+			req: &pb.BeaconBlocksByRangeRequest{
+				Step:  rangeLimit + 100,
+				Count: 1,
+			},
+			expectedError: reqError,
+			errorToLog:    "validation did not fail with bad step",
+		},
+		{
+			name: "Correct Step",
+			req: &pb.BeaconBlocksByRangeRequest{
+				Step:  rangeLimit - 100,
+				Count: 2,
+			},
+			errorToLog: "validation failed with correct step",
+		},
+		{
+			name: "Over Limit Start Slot",
+			req: &pb.BeaconBlocksByRangeRequest{
+				StartSlot: uint64(slotsSinceGenesis) + 1,
+				Step:      1,
+				Count:     1,
+			},
+			expectedError: reqError,
+			errorToLog:    "validation did not fail with bad start slot",
+		},
+		{
+			name: "Over Limit End Slot",
+			req: &pb.BeaconBlocksByRangeRequest{
+				Step:  1,
+				Count: uint64(slotsSinceGenesis) + 10,
+			},
+			expectedError: reqError,
+			errorToLog:    "validation did not fail with bad end slot",
+		},
+		{
+			name: "Exceed Range Limit",
+			req: &pb.BeaconBlocksByRangeRequest{
+				Step:  3,
+				Count: uint64(slotsSinceGenesis / 2),
+			},
+			expectedError: reqError,
+			errorToLog:    "validation did not fail with bad range",
+		},
+		{
+			name: "Valid Request",
+			req: &pb.BeaconBlocksByRangeRequest{
+				Step:      1,
+				Count:     uint64(params.BeaconNetworkConfig().MaxRequestBlocks) - 100,
```

Comment:
If it works with `-1`, it will work with `-100`, so more tight testing conditions are suggested here as well.